src/plugins/intel_cpu/src/emitters/snippets/x64/jit_brgemm_copy_b_emitter.cpp

```diff
@@ -128,15 +199,26 @@ void jit_brgemm_copy_b_emitter::emit_impl(const std::vector<size_t>& in, const s
     Xbyak::Reg64 comp(static_cast<int>(m_with_comp ? out[1] : 0));
 
     const size_t data_size = m_brg_weight_etype.size();
-    for (size_t nb = 0; nb < div_up(m_N_blk, m_inner_N_block); nb++) {
-        const size_t offset_in = m_in_offset + nb * m_inner_N_block * data_size;
-        const size_t offset_out = m_out_offset + nb * m_inner_N_block * m_brgemmVNNIFactor * data_size;
-        const size_t offset_comp = m_with_comp ? m_comp_offset + nb * m_inner_N_block * sizeof(int32_t) : 0;
-
-        const bool is_N_tail = (m_N_blk - nb * m_inner_N_block < m_inner_N_block);
-        const auto current_N_blk = is_N_tail ? m_inner_N_tail : m_inner_N_block;
+    size_t start_in = m_in_offset;
+    size_t start_out = m_out_offset;
+    size_t start_comp = m_comp_offset;
+
+    // OneDNN requires tail handling before main iterations
+    if (m_inner_N_tail != 0) {
+        emit_kernel_call(m_kernel.get(), src, dst, comp, m_inner_N_tail, m_K_blk, start_in, start_out, start_comp);
+        start_in += m_transpose ? m_K * m_inner_N_tail * data_size : m_inner_N_tail * data_size;
```

Comment:
Here and also on L214:
It looks like we really support two very specific data layouts, then why don't we check for these particular layouts on L108?
`m_transpose = !layout.empty() && layout.back() != layout.size() - 1;`
Isn't this check too vague?

Reply:
You are right: N dims can be only last or pre last dim in the layout. I added the corresponding check in constructor